lib/ipam/ipam_test.go

```diff
@@ -90,14 +100,92 @@ type testArgsClaimAff struct {
 
 var _ = testutils.E2eDatastoreDescribe("IPAM tests", testutils.DatastoreEtcdV3, func(config apiconfig.CalicoAPIConfig) {
 	// Create a new backend client and an IPAM Client using the IP Pools Accessor.
-	// Tests that need to ensure a clean datastore should invokke Clean() on the datastore at the start of the
+	// Tests that need to ensure a clean datastore should invoke Clean() on the datastore at the start of the
 	// tests.
 	bc, err := backend.NewClient(config)
 	if err != nil {
 		panic(err)
 	}
 	ic := NewIPAMClient(bc, ipPools)
 
+	Context("With a clean datastore", func() {
```

Comment:
I think the context here should be something that describes the tests - e.g. "Measuring allocation performance"